internal/daemon/controller/handler.go

```diff
@@ -737,3 +738,20 @@ func wrapHandlerWithCallbackInterceptor(h http.Handler, c *Controller) http.Hand
 		h.ServeHTTP(w, req)
 	})
 }
+
+// getActions takes in a URL Path and returns the actions from the URL
+func getActions(path string) []string {
+	// Remove any query parameters
+	path = strings.Split(path, "?")[0]
+
+	parts := strings.Split(path, "/")
+	lastPart := parts[len(parts)-1]
```

Comment:
If this is a slash separated path, you can use `path.Base` to get the last element. You'll need to rename the `path` variable to avoid shadowing the `path` standard library package.
```suggestion
	lastPart := path.Base(urlPath)
```